docs/deployment-examples/custom-scripts/index.md

```diff
@@ -4,112 +4,106 @@ description: Custom scripts allows you to script anything you want using PowerSh
+ - Your scripts can be pre-authenticated and bootstrapped into an external service or server cluster. Learn about [Kubernetes deployments](/docs/deployment-examples/kubernetes-deployments/index.md) and [Service Fabric deployments](/docs/deployment-examples/azure-deployments/service-fabric/index.md).
+ - You can define reusable functions for your scripts to use. Learn about [script modules](docs/deployment-examples/custom-scripts/script-modules.md).
 
-If you're executing a script contained within a package, the package contents will be uncompressed and copied to this directory, but the working directory is the directory containing the script within it.
+## How To Use Custom Scripts
 
-## Collecting Artifacts {#Customscripts-Collectingartifacts}
+Octopus supports the following ways to use custom scripts:
 
-If your deployment produce a log file, configuration files, binaries, or test results that you want to publish and keep as part of your deployment, your scripts can instruct the Octopus Server to collect files as deployment [artifacts](/docs/deployment-process/artifacts.md).
+ - You can define a step in a process to run a script. Learn about the [run a script step](run-a-script-step.md).
+ - You can run custom scripts when Octopus deploys a package. Learn about the [deploy a package step](/docs/deployment-examples/package-deployments/index.md) and [the different stages where your script can run](/docs/deployment-examples/package-deployments/package-deployment-feature-ordering.md).
+ - You can build reusable step templates containing your own scripts which can be used across multiple projects. Learn about [step templates](/docs/deployment-process/steps/custom-step-templates.md).
+ - You can run ad hoc scripts for administrative tasks. Learn about the [script console](/docs/administration/managing-infrastructure/script-console.md).
+ - You can run custom scripts as part of a health check. Learn about [machine policies](/docs/infrastructure/deployment-targets/machine-policies.md).
 
-```powershell PowerShell
-New-OctopusArtifact -Path "C:\Windows\System32\drivers\etc\hosts" -Name "$([System.Environment]::MachineName)-hosts.txt"
-```
+## Where to Store Your Scripts
 
-```cs C#
-Octopus.CreateArtifact(@"C:\Windows\System32\drivers\etc\hosts", System.Environment.MachineName + "-hosts.txt");
-```
+Octopus can execute scripts from a variety of locations, all with different benefits:
 
-```bash Bash
-new_octopusartifact /etc/hosts $(hostname)-hosts.txt
-```
+  1. You can author and store your scripts directly in Octopus. Learn about the [run a script step](run-a-script-step.md) and [deploy a package step](/docs/deployment-examples/package-deployments/index.md).
+  2. You can author your scripts and publish them as step templates for other projects to use. Learn about [step templates](/docs/deployment-process/steps/custom-step-templates.md).
```

Comment:
Is it worth mentioning that these are also stored in Octopus?

Reply:
There are ways people can author and manage their step templates in source control as well. I might leave that detail for the step templates area.